system76/gaze18/default.nix

```diff
@@ -0,0 +1,37 @@
+{ config, lib, pkgs, ... }:
+{
+  imports = [
+    ../.
+  ];
+  boot.initrd.kernelModules = [ "nvidia" ];
+
+  hardware.opengl = {
+    enable = lib.mkDefault true;
+    driSupport = lib.mkDefault true;
+    driSupport32Bit = lib.mkDefault true;
+    extraPackages = with pkgs; [
+      vaapiVdpau
+    ];
+  };
+
+  services.xserver.videoDrivers = ["nvidia"];
+
+  hardware.nvidia = {
+
+    modesetting.enable = lib.mkDefault true;
```

Comment:
From the documentation option:

> This is not enabled by default because it is not officially supported by NVIDIA and would not work with SLI

Can you add a comment why we are enabling it here?

Reply:
I am enabling it because the [NixOS Nvidia wiki](https://wiki.nixos.org/wiki/Nvidia) says to.
```
  hardware.nvidia = {

    # Modesetting is required.
    modesetting.enable = true;
```

If it's not required I can remove it.

Reply:
It further says:

```
Enabling this fixes screen tearing when using Optimus via PRIME (see hardware.nvidia.prime.sync.enable.
```

Did you notice this when enabling/disabling this option?

Reply:
prime sync didn't work at all, when I enabled it and rebooted I couldn't get a graphical desktop, I had to open a tty and disable it.

I haven't noticed any screen tearing, but I also haven't really used the nvidia card, I assume KDE has been running on the integrated graphics, and I haven't done anything graphically intensive because I can't get vulkan to work, so I haven't really been able to put anything to the test, also I've had it enabled since the start because I copied and pasted it when adding nvidia drivers to my system configuration. I can try without it later today.

Reply:
Just tried it and now the nvidia driver isn't being loaded, one of the changes from this morning probably caused this.

Reply:
removing modesetting had no visible impact so far.